idaes/core/util/model_serializer.py

```diff
@@ -701,8 +729,7 @@ def _read_component_data(sd, o, wts, lookup=None, suffixes=None):
     except AttributeError:
         item_keys = [None]
     for key in item_keys:
-        if key is None and isinstance(o, ComponentData) \
-            and not isinstance(o, Component):
+        if key is None and isinstance(o, ComponentData):
             el = o
         else:
             el = o[key]
```

Comment:
Technically, you shouldn't need this test: `o[None] == o` for Scalar components.  (same comment applies to line 518)

Reply:
Yeah, I shouldn't need to check this.  The thing is I sort of messed this up, if a component doesn't have keys, I pretend that it does and put the data under the None index for some sort of consistency.  I newly discovered that FiniteSetOf is a component but apparently doesn't have keys.  I guess I should probably just skip if there are no keys.  If it's a FiniteSetOf or the like, I don't have any need to write it.  I guess this was just me not being able to decide how to handle this.